In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import re


In [8]:
datos  = pd.read_csv('Complaint_Attorney_2023_CDMX.csv')
datos

idCarpeta  Año_inicio Mes_inicio FechaInicio   
0        9361878        2023      Enero  2023-01-01  \
1        9361879        2023      Enero  2023-01-01   
2        9361880        2023      Enero  2023-01-01   
3        9361881        2023      Enero  2023-01-01   
4        9361882        2023      Enero  2023-01-01   
...          ...         ...        ...         ...   
64339    9429532        2023      Marzo  2023-03-31   
64340    9429533        2023      Marzo  2023-03-31   
64341    9429534        2023      Marzo  2023-03-31   
64342    9429535        2023      Marzo  2023-03-31   
64343    9429536        2023      Marzo  2023-03-31   

                                                  Delito   
0      LESIONES CULPOSAS POR TRANSITO VEHICULAR EN CO...  \
1         ROBO A TRANSEUNTE EN VIA PUBLICA CON VIOLENCIA   
2      ROBO DE VEHICULO DE SERVICIO PARTICULAR CON VI...   
3      LESIONES CULPOSAS POR TRANSITO VEHICULAR EN CO...   
4                    PERDIDA DE LA VIDA POR OTRAS CAUSAS   
...                                                  ...   
64339  ROBO S/V DENTRO DE NEGOCIOS, AUTOSERVICIOS, CO...   
64340                                 VIOLENCIA FAMILIAR   
64341        ROBO DE OBJETOS DEL INTERIOR DE UN VEHICULO   
64342                                    ROBO DE OBJETOS   
64343                                 VIOLENCIA FAMILIAR   

                                               Categoria       Sexo  Edad   
0                                 DELITO DE BAJO IMPACTO        NaN   NaN  \
1      ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIO...  Masculino  50.0   
2                   ROBO DE VEHÍCULO CON Y SIN VIOLENCIA  Masculino  34.0   
3                                 DELITO DE BAJO IMPACTO  Masculino   NaN   
4                                     HECHO NO DELICTIVO  Masculino  90.0   
...                                                  ...        ...   ...   
64339                             DELITO DE BAJO IMPACTO   Femenino  69.0   
64340                             DELITO DE BAJO IMPACTO   Femenino  17.0   
64341                             DELITO DE BAJO IMPACTO   Femenino  43.0   
64342                             DELITO DE BAJO IMPACTO   Femenino  39.0   
64343                             DELITO DE BAJO IMPACTO   Femenino  41.0   

      TipoPersona        CalidadJuridica  ...  Mes_hecho  FechaHecho   
0           MORAL               OFENDIDO  ...  Diciembre  2022-12-31  \
1          FISICA  VICTIMA Y DENUNCIANTE  ...  Diciembre  2022-12-31   
2          FISICA  VICTIMA Y DENUNCIANTE  ...  Diciembre  2022-12-31   
3          FISICA              LESIONADO  ...  Diciembre  2022-12-31   
4          FISICA                CADAVER  ...  Diciembre  2022-12-31   
...           ...                    ...  ...        ...         ...   
64339      FISICA  VICTIMA Y DENUNCIANTE  ...      Marzo  2023-03-31   
64340      FISICA                VICTIMA  ...      Marzo  2023-03-30   
64341      FISICA  VICTIMA Y DENUNCIANTE  ...      Marzo  2023-03-31   
64342      FISICA  VICTIMA Y DENUNCIANTE  ...      Marzo  2023-03-31   
64343      FISICA  VICTIMA Y DENUNCIANTE  ...      Marzo  2023-03-31   

      HoraHecho HoraInicio      alcaldia_hechos municipio_hechos   
0      22:08:00   00:09:00             COYOACAN              NaN  \
1      23:50:00   00:41:00  VENUSTIANO CARRANZA              NaN   
2      20:30:00   00:44:00    GUSTAVO A. MADERO              NaN   
3      23:30:00   00:47:00             COYOACAN              NaN   
4      10:30:00   00:56:00    GUSTAVO A. MADERO              NaN   
...         ...        ...                  ...              ...   
64339  07:30:00   23:48:00        BENITO JUAREZ              NaN   
64340  22:00:00   23:49:00           IZTAPALAPA              NaN   
64341  07:00:00   23:51:00       ALVARO OBREGON              NaN   
64342  10:01:00   23:54:00              TLALPAN              NaN   
64343  17:30:00   23:55:00         AZCAPOTZALCO              NaN   

                  

In [9]:
datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64344 entries, 0 to 64343
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   idCarpeta             64344 non-null  int64  
 1   Año_inicio            64344 non-null  int64  
 2   Mes_inicio            64344 non-null  object 
 3   FechaInicio           64344 non-null  object 
 4   Delito                64344 non-null  object 
 5   Categoria             64344 non-null  object 
 6   Sexo                  54929 non-null  object 
 7   Edad                  44981 non-null  float64
 8   TipoPersona           64212 non-null  object 
 9   CalidadJuridica       64344 non-null  object 
 10  competencia           64344 non-null  object 
 11  Año_hecho             64337 non-null  float64
 12  Mes_hecho             64337 non-null  object 
 13  FechaHecho            64337 non-null  object 
 14  HoraHecho             64337 non-null  object 
 15  HoraInicio         

Tenemos un total de 64344 patrones, con 16 clases y un IR de: 93.93284671532847

In [12]:
ir = float(len(datos["Categoria"])/datos["municipio_hechos"].count())
print(ir)

93.93284671532847
